tea.py

```diff
@@ -0,0 +1,674 @@
+                pe_composite_wt_fraction,
+            )
+            print("Separator composite weight fraction: ", sep_composite_wt_fraction)
+            print(
+                "Negative electrode with electrolyte density [kg.m-3]: ",
+                ne_w_elyte_density,
+            )
+            print(
+                "Positive electrode with electrolyte density [kg.m-3]: ",
+                pe_w_elyte_density,
+            )
+            print("Separator with electrolyte density [kg.m-3]: ", sep_w_elyte_density)
+            print(
+                "Negative electrode electrolyte weight fraction: ", ne_elyte_wt_fraction
+            )
+            print(
+                "Positive electrode electrolyte weight fraction: ", pe_elyte_wt_fraction
+            )
+            print("Separator electrolyte weight fraction: ", sep_elyte_wt_fraction)
+            print("Electrolyte density [kg.m-3]: ", elyte_density)
+            print("Negative electrode porosity: ", ne_porosity)
+            print("Positive electrode porosity: ", pe_porosity)
+            print("Separator porosity: ", sep_porosity)
+            print(
+                "Negative electrode with electrolyte density [kg.m-3]: ",
+                ne_w_elyte_density,
+            )
+            print(
+                "Positive electrode with electrolyte density [kg.m-3]: ",
+                pe_w_elyte_density,
+            )
+            print("Separator with electrolyte density [kg.m-3]: ", sep_w_elyte_density)
+            print(
+                "Negative electrode electrolyte weight fraction: ", ne_elyte_wt_fraction
+            )
+            print(
+                "Positive electrode electrolyte weight fraction: ", pe_elyte_wt_fraction
+            )
+            print("Separator electrolyte weight fraction: ", sep_elyte_wt_fraction)
+            print("Electrolyte density [kg.m-3]: ", elyte_density)
+            print(
+                "Negative electrode composite density [kg.m-3]: ", ne_composite_density
+            )
+            print(
+                "Positive electrode composite density [kg.m-3]: ", pe_composite_density
+            )
+            print("Separator composite density [kg.m-3]: ", sep_composite_density)
+            print(
+                "Negative electrode composite weight fraction: ",
+                ne_composite_wt_fraction,
+            )
+            print(
+                "Positive electrode composite weight fraction: ",
+                pe_composite_wt_fraction,
+            )
+            print("Separator composite weight fraction: ", sep_composite_wt_fraction)
+
+        return volumetric_stack_energy_density, gravimetric_stack_energy_density
+
+    def stack_breakdown(self, input_parameter=None, print_values=False):
+        # Assign densities based on ocv function and set electrolyte density to 1300 kg m-3?
+        parameter_values = self.parameter_values
+        if input_parameter is None:
```

Comment:
For consistency with PyBaMM let's call this `inputs` everywhere, and then do `inputs = inputs or {}` here